server/api/core/core.js

```diff
@@ -163,6 +163,12 @@ export default {
     return shop && shop._id;
   },
 
+  // Use the current user Id to get the shop Id
+  getVendorId(userId) {
+    const shop = Shops.find({shopOwnerId: userId}).fetch()[0];
+    return (shop) ? shop && shop._id : false; // Return the shop details or false.
```

Comment:
Here you are checking if `shop` exists multiple times. you should either have:
```js
return (shop) ? shop._id : false;
```
or
```js
return shop && shop._id;
```